README.md
## Gradient Box

A **package** that lets you add a beautiful modifiable **Gradient Box** to your Flutter app.

&nbsp;

## Installation

1. Add the latest version of package to your pubspec.yaml (and run `dart pub get` ):

```yaml
dependencies:
  gradientbox: ^0.0.2
```

2. Import the package and use it in your Flutter App.

```dart
import 'package:gradientbox/gradientbox.dart';
```

&nbsp;

## Getting Started

There are a number of properties that you can modify:

- onTap
- height
- width
- padding
- margin
- alignment
- borderRadius
- showShadow
- shadowColor
- gradientColors
- child

<hr>
<table>
<tr>
<td>

```dart
class GradientBoxDemoScreen extends StatelessWidget {
  const GradientBoxDemoScreen({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        backgroundColor: Colors.white,
        appBar: AppBar(
          title: const Text(
            "Demo 💯",
            style: TextStyle(color: Colors.black, fontSize: 14),
          ),
          centerTitle: true,
          backgroundColor: Colors.white,
          elevation: 0.5,
        ),
        body: Column(
          children: [
            Expanded(
              child: Center(
                child: GradientBox(
                    margin: const EdgeInsets.all(20),
                    padding: const EdgeInsets.symmetric(
                        vertical: 25, horizontal: 30),
                    height: MediaQuery.of(context).size.height * 0.25,
                    borderRadius: 15,
                    gradientColors: GradientColors.blueSkies,
                    alignment: Alignment.topLeft,
                    showShadow: true,
                    shadowColor: Colors.grey,
                    onTap: () =>
                        ScaffoldMessenger.of(context).showSnackBar(
                        const SnackBar(content: Text("Blue Skies"))),
                    child: content),
              ),
            ),
            const Padding(
              padding: EdgeInsets.all(15.0),
              child: Text(
                "GradientBox",
                style: TextStyle(fontSize: 10, color: Colors.blue),
              ),
            )
          ],
        ));
  }
}
```

</td>
<td>
<img src="https://i.ibb.co/T8mGjLS/Screenshot-1655502524.png" width="100%" height="100%"  alt="demo" border="0" />
</td>
</tr>
</table>

&nbsp;

## Contributing

Gradient Box is **100% free** and **open source**. This package accepts contributions from the public &ndash; including you. There are a couple of ways in which you can contribute to the growing community of `GradientBox`.

- **Add gradient colors to the package.**

1. To add your gradient colors, fork this repository, then go to `gradient_colors.dart` file, and add a `GradientColors` variable which takes a list of colors (All the gradient colors are loaded from a single `gradient_colors.dart` file in `\lib\utils\`).

```dart
class GradientColors {

  ...

  static GradientColors flare = GradientColors(const [
    Color(0xFFf12711),
    Color(0xFFf5af19),
  ]);

}

```

2. After that add your `GradientColors` variable to the `_gradientColors` list and then submit a pull request.

```dart
class GradientColors {

  ...

  static final List<GradientColors> _gradientColors = [
    ...

    flare,
   ]
}

```

- **Fix a bug**
- **If you are a developer, feel free to check out the source and submit pull requests.**

_NOTE_ - Please keep gradient submissions and bug fixes in separate PRs.

&nbsp;

### ❤️ Found this project useful?

If you found this project useful, then please consider giving it a ⭐ on Github and sharing it with your friends.

&nbsp;

<p align="center">✌️</p>
<p align="center">
<sub><sup>A little project by <a href="https://www.instagram.com/aquib.hamid/">@aquib.hamid</a></sup></sub>
</p>


CHANGELOG.md
## 0.0.1

Added the modifiable Gradient Box.

## 0.0.2

Updated Dart SDK


lib/src/gradientbox.dart
import 'package:flutter/material.dart';
import '../gradientbox.dart';

class GradientBox extends StatelessWidget {
  /// If non-null, then defines what will happen on tapping the widget.
  final VoidCallback? onTap;

  /// The [child] contained by the Gradient Box.
  ///
  /// If null, a `Text("Hello World")` widget will be added automatically
  /// and Gradient Box will expand to fill all available space in its parent.
  ///
  /// If the parent provides unbounded constraints, in which case the box
  /// will attempt to be as small as possible.
  ///
  /// This widget can only have one child. To lay out multiple children, let
  /// this widget's child be a widget such as [Row], [Column], or [Stack],
  /// which have a children property. and then provide the children to
  /// that widget.
  final Widget? child;

  final double? height;

  final double? width;

  /// If non-zero, the corners of this widget are rounder by the provide value.
  ///
  /// If null, then the default value is 25.
  final double? borderRadius;

  /// Creates a Linear gradient.
  ///
  /// If non-null, the [colors] argument must be passed in `GradientColors()` .
  ///
  /// If null, then random gradient colors are chosen.
  final GradientColors? gradientColors;

  final bool showShadow;

  /// Adds a shadow to the widget. [showShadow] must to true to see the effect.
  ///
  /// If null, then the default color is `colors.grey`.
  final Color? shadowColor;

  /// Align the [child] within the box.
  ///
  /// If non-null, the Gradient Box will expand to fill its parent and position its
  /// child within itself according to the given value. If the incoming
  /// constraints are unbounded, then the child will be shrink-wrapped instead.
  ///
  /// If null, then will have a default value of `Alignment.center`.
  ///
  /// Ignored if [child] is null.
  final AlignmentGeometry? alignment;

  /// Empty space surrounding the [child].
  final EdgeInsets? padding;

  /// Empty space surrounding the widget.
  final EdgeInsets? margin;

  ///Create a customizable Gradient Box widget with a beautiful gradient.
  const GradientBox(
      {Key? key,
      this.onTap,
      this.alignment,
      this.height,
      this.width,
      this.padding,
      this.margin,
      this.gradientColors,
      this.borderRadius,
      this.showShadow = true,
      this.shadowColor,
      this.child})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return InkWell(
      onTap: onTap ?? () {},
      splashColor: Colors.transparent,
      highlightColor: Colors.transparent,
      enableFeedback: onTap == null ? false : true,
      child: Container(
        alignment: alignment ?? Alignment.center,
        padding: padding,
        margin: margin,
        height: height ?? MediaQuery.of(context).size.height * 0.25,
        width: width,
        decoration: BoxDecoration(
          borderRadius: BorderRadius.all(Radius.circular(borderRadius ?? 25.0)),
          gradient: LinearGradient(
              colors: gradientColors == null
                  ? GradientColors.random().colors
                  : gradientColors!.colors,
              begin: Alignment.topLeft,
              end: Alignment.bottomRight),
          boxShadow: showShadow
              ? [
                  BoxShadow(
                    color: shadowColor ?? Colors.grey,
                    blurRadius: 20,
                    offset: const Offset(0, 6),
                  ),
                ]
              : [],
        ),
        child: child ??
            const Text(
              "Hello World",
              style: TextStyle(
                  fontSize: 16,
                  fontWeight: FontWeight.bold,
                  color: Colors.white),
            ),
      ),
    );
  }
}


lib/utils/gradient_colors.dart
import 'dart:math';
import 'dart:ui';

class GradientColors {
  final List<Color> colors;
  GradientColors(this.colors);

  factory GradientColors.random() {
    return _gradientColors[Random().nextInt(_gradientColors.length)];
  }

  // After adding GradientColors variable, add that variable
  // to this list as shown below:

  static final List<GradientColors> _gradientColors = [
    flare,
    summer,
    cosmicFusion,
    radioactiveHeat,
    iceToFire,
    skySea,
    harvey,
    scooter,
    newBlu,
    witchingHour,
    ultraViolet,
    burningOrange,
    wireTap,
    blueRaspberry,
    sublimeLight,
    lawrencium,
    argon,
    telegram,
    blueSkies,
    coal,
    html,
    pacificGreen,
    minimalRed,
  ];

  // Don't forget commas and semi-colons...and try to follow the format.
  //
  // Add GradientColors variable below:
  //

  static GradientColors flare = GradientColors(const [
    Color(0xFFf12711),
    Color(0xFFf5af19),
  ]);
  static GradientColors minimalRed = GradientColors(const [
    Color(0xFFf00000),
    Color(0xFFdc281e),
  ]);
  static GradientColors summer = GradientColors(const [
    Color(0xFFF44336),
    Color(0xFFFFC107),
    Color(0xFF4CAF50),
  ]);
  static GradientColors cosmicFusion = GradientColors(const [
    Color(0xFFff00cc),
    Color(0xFF333399),
  ]);
  static GradientColors iceToFire = GradientColors(const [
    Color(0xFF72C6EF),
    Color(0xFF004E8F),
  ]);
  static GradientColors radioactiveHeat = GradientColors(const [
    Color(0xFFF7941E),
    Color(0xFF72C6EF),
    Color(0xFF00A651),
  ]);
  static GradientColors skySea = GradientColors(const [
    Color(0xFFE4E5E6),
    Color(0xFF00416A),
  ]);
  static GradientColors harvey = GradientColors(const [
    Color(0xFF1f4037),
    Color(0xFF99f2c8),
  ]);
  static GradientColors scooter = GradientColors(const [
    Color(0xFF7F7FD5),
    Color(0xFF86A8E7),
    Color(0xFF91EAE4),
  ]);
  static GradientColors witchingHour = GradientColors(const [
    Color(0xFFC31432),
    Color(0xFF240b36),
  ]);
  static GradientColors ultraViolet = GradientColors(const [
    Color(0xFF654ea3),
    Color(0xFFeaafc8),
  ]);
  static GradientColors burningOrange = GradientColors(const [
    Color(0xFFFF416C),
    Color(0xFFFF4B2B),
  ]);
  static GradientColors wireTap = GradientColors(const [
    Color(0xFF8A2387),
    Color(0xFFE94057),
    Color(0xFFF27121),
  ]);
  static GradientColors blueRaspberry = GradientColors(const [
    Color(0xFF00B4DB),
    Color(0xFF0083B0),
  ]);
  static GradientColors sublimeLight = GradientColors(const [
    Color(0xFFFC5C7D),
    Color(0xFF6A82FB),
  ]);
  static GradientColors lawrencium = GradientColors(const [
    Color(0xFF0f0c29),
    Color(0xFF302b63),
    Color(0xFF24243e),
  ]);
  static GradientColors argon = GradientColors(const [
    Color(0xFF03001e),
    Color(0xFF7303c0),
    Color(0xFFec38bc),
    Color(0xFFfdeff9),
  ]);
  static GradientColors rainbowBlue = GradientColors(const [
    Color(0xFF00F260),
    Color(0xFF0575E6),
  ]);
  static GradientColors newBlu = GradientColors(const [
    Color(0xFF396afc),
    Color(0xFF2948ff),
  ]);
  static GradientColors telegram = GradientColors(const [
    Color(0xFF1c92d2),
    Color(0xFFf2fcfe),
  ]);
  static GradientColors blueSkies = GradientColors(const [
    Color(0xFF56CCF2),
    Color(0xFF2F80ED),
  ]);
  static GradientColors coal = GradientColors(const [
    Color(0xFFEB5757),
    Color(0xFF000000),
  ]);
  static GradientColors html = GradientColors(const [
    Color(0xFFE44D26),
    Color(0xFFF16529),
  ]);
  static GradientColors pacificGreen = GradientColors(const [
    Color(0xFF34e89e),
    Color(0xFF0f3443),
  ]);
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:gradientbox/gradientbox.dart';

class DemoApp extends StatelessWidget {
  const DemoApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) => const MaterialApp(
        title: "Example on how to use GradientBox",
        home: DemoScreen(),
      );
}

class DemoScreen extends StatelessWidget {
  const DemoScreen({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) => Scaffold(
        appBar: AppBar(
          title: const Text(
            "Demo 💯",
            style: TextStyle(color: Colors.black, fontSize: 14),
          ),
          centerTitle: true,
          backgroundColor: Colors.white,
          elevation: 0.5,
        ),
        body: ListView(
          children: [
            //Gradient Box with random gradient colors.
            GradientBox(
              onTap: () => ScaffoldMessenger.of(context).showSnackBar(
                  const SnackBar(
                      content: Text("You forgot to add gradient colors ? 😅"))),
              margin: const EdgeInsets.all(20),
            ),

            // You can provide your desired colors to `GradientColors()`.
            GradientBox(
              margin: const EdgeInsets.all(20),
              gradientColors: GradientColors(const [
                Color(0xFF00B4DB),
                Color(0xFF0083B0),
              ]),
              child: const Text(
                "Created with Dart 🎯",
                style: TextStyle(
                    fontSize: 16,
                    fontWeight: FontWeight.bold,
                    color: Colors.white),
              ),
            ),

            // You can also use predefined gradient colors that comes with package.
            GradientBox(
              margin: const EdgeInsets.all(20),
              gradientColors: GradientColors.burningOrange,
              child: const Text(
                "Orange Lover 🍊",
                style: TextStyle(
                    fontSize: 16,
                    fontWeight: FontWeight.bold,
                    color: Colors.white),
              ),
            ),

            // You can modify your Gradient Box however you want.
            GradientBox(
              margin: const EdgeInsets.all(20),
              gradientColors: GradientColors(const [
                Color(0xFFFF3CAC),
                Color(0xFF784BA0),
                Color(0xFF2B86C5),
              ]),
              borderRadius: 100,
              height: 120,
              showShadow: true,
              shadowColor: const Color(0xFFFF3CAC),
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: const [
                  Text(
                    "GradientContainer",
                    style: TextStyle(
                        fontSize: 20,
                        fontWeight: FontWeight.bold,
                        color: Colors.white),
                  ),
                  SizedBox(
                    height: 20,
                  ),
                  Text(
                    "Easy to create gradient containers ✌️",
                    style: TextStyle(
                        fontSize: 14,
                        fontWeight: FontWeight.w500,
                        color: Colors.white),
                  )
                ],
              ),
            ),
            const Padding(
              padding: EdgeInsets.only(top: 10.0, bottom: 20.0),
              child: Center(
                  child: Text(
                "GradientContainer",
                style: TextStyle(fontSize: 12, color: Colors.black45),
              )),
            )
          ],
        ),
      );
}
